sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -1569,6 +1591,72 @@ case class GpuLast(child: Expression, ignoreNulls: Boolean)
       TypeCheckSuccess
     }
   }
+
+  // GENERAL WINDOW FUNCTION
+  override lazy val windowInputProjection: Seq[Expression] = inputProjection
+  override def windowAggregation(
+      inputs: Seq[(ColumnVector, Int)]): RollingAggregationOnColumn =
+    RollingAggregation.nth(-1, if (ignoreNulls) NullPolicy.EXCLUDE else NullPolicy.INCLUDE)
+        .onColumn(inputs.head._2)
+}
+
+case class GpuNthValue(child: Expression, offset: Expression, ignoreNulls: Boolean)
+    extends GpuAggregateFunction
```

Comment:
From what I have seen in the code and docs for Spark NthValue is only supported for window functions. So we should not support GpuAggregateFunction. This should look more like GpuLead or GpuLag. This is very similar.